frontend/device/devicelistener.lua

```diff
@@ -202,10 +202,18 @@ if Device:hasFrontlight() then
             direction = 1
         end
         local warmth = powerd.fl_warmth + direction * delta_int
-        if warmth > 100 then
```

Comment:
I'm not familiar with devicelistener. Pinging @yparitcher to make sure we don't clash with other warmth devices

Reply:
TBH, i am not very familiar with the warmth code, I don't have a warmth device, and the code was moved from `readerfrontlight` which is originally from `kobolight.koplugin`.

However from looking around, the frontend expects warmth to be 1 .. 100 (in frontlightwidget etc) and that powerd should translate that to whatever scale the device actually uses. so either have luajit-launcher do this for android, or fix that assumption everywhere.

pinging @NiLuJe & @Frenzie who should actually know what they are talking about. ;)

Reply:
Yeah, the front-end expects a 0-100 range, and it's the backend's responsibility to scale that appropriately.

Reply:
e.g., https://github.com/koreader/koreader/blob/db843929f0893383a3dcddbb6d77708691b8e393/frontend/device/kobo/powerd.lua#L47-L50

Some devices used a Kelvin scale on their end, others use an arbitrary [0, ??] range, or whatever else they want, really ^^.

(i.e., it's a mess :D).

Reply:
@NiLuJe: The Tolino Epos2, on which I am developing should have almost the same Hardware as the Kobo Forma (but has an android firmware). Therefore, it should also use (direct or through the firmware) the same `/sys/class/backlight/tlc5947_bl/color` file.

I know nothing about the Kobo. Could it be, that also the kobo has only 10 steps of warmth and that the "color temperature" is just calculated somehow within the kobo firmware? Do you notice steps, when you change the warmth of the light? How many steps?

Reply:
That excerpt was for the Aura One, which had fancier (and buggier) LEDs.

Newer devices do rely on a hardware mixer which only exposes a [0, 10] range:

https://github.com/koreader/koreader/blob/db843929f0893383a3dcddbb6d77708691b8e393/frontend/device/kobo/device.lua#L231-L239

Reply:
Now I see. Thank you.

Reply:
@NiLuJe: When I look at the `devicelistener.lua` I realize, 
1.) that there is a scale for the intensity, so that you can swipe from `fl_min` to `fl_max`. 
2.) For warmth there is a fixed scale from 0 to 100.

Especially for my device, where I have only 11 steps of warmth a scale from 0..100 is annoying. 
Are there reasons, why the 100-scale should be there. Or shall we change it?

Reply:
No idea, I'm not the one who initially implemented warmth ;).

Since the first one was probably the KA1, which happened to be extremely flexible, a [0, 100] range probably felt like a good user-facing scale.

IIRC, I simply hacked the scaling for the NL widget.

I don't particularly care about the gesture, but I don't really see a great way to handle it either, it's already super vague to begin with, so it works well enough as-is.